dataprofiler/profilers/json_decoder.py

```diff
@@ -0,0 +1,65 @@
+"""Contains methods to decode components of a Profiler."""
```

Comment:
I think we might want to consider a slightly separate API that implements functionality in the profiles themselves.

e.g.
```python.

# presuming i have this file already saved as JSON.
filepath = <LOCATION_OF_DATA>
profiler = Profiler.load(filepath, type="json")

# or 
profiler = Profiler.load_json(filepath)
```

Within each profile we can handle the profiler load checks,
for categorical it could be something like:
```python
class Cat...Profiler(...):
    .
    .
    .
    @classmethod
    def load_json_file(cls, filepath):
        ...

    @classmethod
    def load_json(cls, json_data):
        if json_data.get("class", None) is not cls.__name__:
            raise ValueError(...)
        profile = cls(...)
        ...
        return profile
```
moreover, ^^ might be able to just exist in BaseColumnProfiler if it is abstract enough.
the `...` part in the load might be able to be private functions in each class the base calls e.g.

```python
   profile = cls(...)
   profile._set_properties(json_data.get("data"))
```
Not keen on `_set_properties` naming, but it would have to exist in each class loading data.

Reply:
or if the code generalizes well enough,  we don't need `set_properties` and just use:
```python
for attr, value in serialized_json["data"].items():
        column_profiler.__setattr__(attr, value)
```

Reply:
I was thinking the same thing. Something like a separate `StructuredColProfile` decode function that calls my `decode_column_profiler` whenever it sees a Column Profiler in its attributes.